Add new classes - CTPPSAlignmentCorrectionsData and PPSAlignmentConfi…

Kszemka opened the pull request:
#### PR description:
The main aim of this PR is to add new plots for CTPPS classes - CTPPSRPAlignmentCorrectionsData and PPSAlignmentConfiguration - to Payload Inspector software so to be able to visualize it on the website.

#### PR validation:
The code was tested by a script testPPSRPAlignmentCorrections.sh (which is mostly based on getPayloadData.py from CondCore/Utilities) just to check whether the code is properly added to PI software and generate some proper data in JSON files produced by PI.
In case of testing - testPPSRPAlignmentCorrections.sh script should be run with 'run' argument.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

No backport.

Kszemka committed:
Add new classes - CTPPSAlignmentCorrectionsData and PPSAlignmentConfiguration for Payload Inspector

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31925

- This PR adds an extra 16KB to repository

cmsbuild commented:
A new Pull Request was created by  @Kszemka (Aleksandra Krzemińska) for master.

It involves the following packages:

- CondCore/CTPPSPlugins (**db**)


@malbouis, @cmsbuild, @saumyaphor4252, @ggovi, @francescobrivio, @tvami can you please review it and eventually sign? Thanks.
@fabferro, @mmusich this is something you requested to watch as well.
@perrotta, @dpiparo, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

tvami commented on CondCore/CTPPSPlugins/interface/CTPPSRPAlignmentCorrectionsDataHelper.h:
```diff
@@ -0,0 +1,109 @@
+    if (r == 3 || r == 23 || r == 103 || r == 123)
+      return r;
+    else {
+      CTPPSDetId* config = new CTPPSDetId(r);
```
is this `delete`-ed?

Kszemka commented on CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:
```diff
@@ -0,0 +1,18 @@
+#include "CondFormats/PPSObjects/interface/PPSAlignmentConfiguration.h"
+#include "CondCore/CTPPSPlugins/interface/PPSAlignmentConfigurationHelper.h"
+
+namespace {
```
In my previous [PR](https://github.com/cms-sw/cmssw/pull/39024) PPSTimingCalibration_PayloadInspector was also formed with unnamed namespace - the same situation involves most of the PI classes, like [SiPixelPlugins](https://github.com/cms-sw/cmssw/blob/master/CondCore/SiPixelPlugins/plugins/SiPixel2DTemplateDBObject_PayloadInspector.cc#L49) or [BeamSpotPlugins](https://github.com/cms-sw/cmssw/blob/master/CondCore/BeamSpotPlugins/plugins/BeamSpotOnline_PayloadInspector.cc#L9) - this is why I wanted to keep these conventions. Should it be changed anyway?

mmusich replied:
I cannot find any reference to having only named namespaces either [here](https://cms-sw.github.io/cms_coding_rules.html) or in the cpp core guidelines. Perhaps @tvami can clarify why that would be needed.

tvami replied:
> or in the cpp core guidelines

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed

Matti reported a problem with this recently here:
https://github.com/cms-sw/cmssw/issues/38977

mmusich replied:
excepted this is not a header...

tvami replied:
ahh yes, that's right

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31945

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #39274 was updated. @malbouis, @cmsbuild, @saumyaphor4252, @ggovi, @francescobrivio, @tvami can you please check and sign again.

Kszemka commented:
The above mentioned issues were fixed :)

tvami commented:
@cmsbuild , please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27291/summary.html
**COMMIT**: 93ea20cba1a038997833109813b33905f89f4537
**CMSSW**: CMSSW_12_6_X_2022-09-01-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39274/27291/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3620448
 - DQMHistoTests: Total failures: 17
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3620408
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 50 files compared)
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 212 log files, 49 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-09-01-1100+c0057b/52534/triggerResults): no differences found

tvami commented:
+db
 * tests pass

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

mmusich commented:
>  tests pass

though the test is actually [not testing anything](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27291/unitTests/src/CondCore/CTPPSPlugins/test/testPPSTimingCalibration/testing.log)

https://github.com/cms-sw/cmssw/blob/a9b836229a8b2869e3d342b70dd1332e45e0681a/CondCore/CTPPSPlugins/test/testPPSTimingCalibration.cpp#L28-L44

why is the test configured like that?

perrotta commented:
@Kszemka please take a look at https://github.com/cms-sw/cmssw/pull/39274#issuecomment-1235601524

Kszemka commented:
The test is working similar to testPPSTimingCalibration.sh but it gives an option to choose tag from the database and runs without changing the code (given as argv[] like this - 
![image](https://user-images.githubusercontent.com/76405587/188420201-d657616f-da63-420f-8ac3-08d4ba42651f.png))... although it was not added in this PR, already merged in one of the previous [ones](https://github.com/cms-sw/cmssw/pull/38915/files#diff-6660e1e20a272eefa2523e65f65348cc7126f626e788883494dc8ef98dfe71d2)

mmusich commented:
can `testPPSTimingCalibration.sh` become part of the unit test suite?

Kszemka commented:
testPPSTimingCalibration.sh also demand some arguments to run it with - maybe the better option is just to change the testPPSTimingCalibration.cpp and also provide inside the code example tag and runs so it will work with _scram b runtests_?

mmusich commented:
> and also provide inside the code example tag and runs so it will work with scram b runtests?

Yes that was my initial comment

Kszemka commented:
Changed testPPSTimingCalibration.cpp so it works with scram b runtests, also added similar testPPSRPAlignmentCorrections.cpp for the new class that was added with this PR.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31992

- This PR adds an extra 12KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31992/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31992/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39274/31993

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #39274 was updated. @malbouis, @cmsbuild, @saumyaphor4252, @ggovi, @francescobrivio, @tvami can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27340/summary.html
**COMMIT**: 5870b207217e678ed71fa544f377567e19088f8c
**CMSSW**: CMSSW_12_6_X_2022-09-05-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/39274/27340/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3618210
 - DQMHistoTests: Total failures: 8
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3618180
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 50 files compared)
 - Checked 212 log files, 49 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-09-05-1100+c0057b/52579/triggerResults): no differences found

tvami commented:
+db
 * unit tests pass https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27340/unitTests/success.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
@Kszemka I'm just curious: why did you name the CTPPSRPAlignmentCorrectionsDataHelper with initial "CTPPS", while the other class introduced here, and the other one already present in CondCore/CTPPSPlugins start with "PPS" (which is, as far as I remember, the agreed naming convention for this code nowadays)?

Kszemka commented:
It is simply connected with the names of the classes that are already used in db and also are used for the PI visualization - there is [CTPPSRPAlignmentCorrectionsData](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/CTPPSRPAlignmentCorrectionsData.cc) and [PPSAlignmentConfiguration](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/PPSAlignmentConfiguration.cc) - thought it would be easier if the names somehow corresponds to each other. Should I change the name of CTPPSRPAlignmentCorrectionsData to PPS at the beginning anyway?

perrotta commented:
> @perrotta, it is simply connected with the names of the classes that are already used in db and also are used for the PI visualization - there is [CTPPSRPAlignmentCorrectionsData](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/CTPPSRPAlignmentCorrectionsData.cc) and [PPSAlignmentConfiguration](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/PPSAlignmentConfiguration.cc) - thought it would be easier if the names somehow correspond to each other. Should I change the name of CTPPSRPAlignmentCorrectionsData to PPS at the beginning anyway?

Ok, it can make sense.
I think that at some point it was decided to migrate all "CTPPS" in the class names into "PPS", and I also think that it should be done at some point. But for now, let stick on the same names of the classes that are already used in db. Until we'll eventually migrate everything it will be easier to match each other as such.

perrotta commented:
+1
